m.neural_network.preparedata.worker_export/m.neural_network.preparedata.worker_export.py

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/env python3
+"""############################################################################
+#
+# MODULE:      m.neural_network.preparedata.worker_export
+# AUTHOR(S):   Guido Riembauer, Anika Weinmann
+# PURPOSE:     Worker module for m.neural_network.preparedata to export data
+# COPYRIGHT:   (C) 2024 by mundialis GmbH & Co. KG and the GRASS Development
+#              Team.
+#
+# 		This program is free software under the GNU General Public
+# 		License (v3). Read the file COPYING that comes with GRASS
+# 		for details.
+#
+#############################################################################
+"""
+
+# %Module
+# % description: Worker module for m.neural_network.preparedata to export data.
+# % keyword: raster
+# % keyword: export
+# % keyword: segmentation
+# %end
+
+# %option G_OPT_R_INPUTS
+# % key: image_bands
+# % label: The names of imagery raster bands, e.g. for DOPs RGBI raster bands
+# % description: The first raster defines the output resolution
+# % guisection: Input
+# %end
+
+# %option G_OPT_R_INPUT
+# % key: ndsm
+# % label: Name of the nDSM raster
+# % answer: ndsm
+# % guisection: Input
+# %end
+
+# %option G_OPT_V_INPUT
+# % key: reference
+# % required: no
+# % label: Name of the reference vector map
+# % guisection: Optional input
+# %end
+
+# %option
+# % key: segmentation_minsize
+# % type: integer
+# % required: no
+# % label: Minimum number of cells in a segment
+# % answer: 80
+# % guisection: Optional input
+# %end
+
+# %option
+# % key: segmentation_threshold
+# % type: double
+# % required: no
+# % label: Difference threshold between 0 and 1 for the segments
+# % description: Threshold = 0 merges only identical segments; threshold = 1 merges all
+# % answer: 0.3
+# % guisection: Optional input
+# %end
+
+# %option G_OPT_M_DIR
+# % key: output_dir
+# % multiple: no
+# % label: Directory where the prepared data should be stored
+# % description: The directory will be split into train and apply
+# % guisection: Output
+# %end
+
+# %option
+# % key: tile_name
+# % type: string
+# % required: yes
+# % multiple: no
+# % key_desc: name
+# % label: Unique Name of the tile
+# %end
+
+# %option
+# % key: new_mapset
+# % type: string
+# % required: yes
+# % multiple: no
+# % label: Name for new mapset
+# %end
+
+# %flag
+# % key: t
+# % label: Export reference or segmentation data as training data suggestion
+# %end
+
+import os
+import shutil
+
+import grass.script as grass
+from grass.pygrass.utils import get_lib_path
+from grass.script.vector import vector_info_topo
+from grass_gis_helpers.mapset import switch_to_new_mapset
+
+EXPORT_PARAM = {
+    "format": "GTiff",
+    "flags": "mc",
+    "createopt": "COMPRESS=LZW,TILED=YES,BIGTIFF=YES",
+    "overviews": 5,
+}
+
```

Comment:
It would be best to have the move back to the original mapset in a cleanup function, such that it also happens in case of an error, see also: https://github.com/mundialis/m.neural_network/pull/2#discussion_r1891181678